users/luescher/setups/hybrid/util.py

```diff
@@ -0,0 +1,421 @@
+__all__ = ["GmmDataInput", "GmmPipelineArgs", "NnArgs"]
+
+import i6_core.meta as meta
+
+
+class GmmDataInput:
+    def __init__(
+        self,
+        corpus_object: meta.CorpusObject = None,
+        lexicon: dict = None,
+        lm: dict = None,
+        concurrent: int = 10,
+    ):
+        """
+        :param corpus_object: corpus_file: Path, audio_dir: Path, audio_format: str, duration: float
+        :param lexicon: file: Path, normalize_pronunciation: bool
+        :param lm: file: Path, type: str, scale: float
+        :param concurrent: concurrency for gmm hmm pipeline
+        """
+        self.corpus_object = corpus_object
+        self.lexicon = lexicon
+        self.lm = lm
+        self.concurrent = concurrent
+
+
+class GmmPipelineArgs:
```

Comment:
Maybe we also want to have each specific args block as own class? I think with this we could document this easier and split things a little bit more...

Reply:
I agree. What granularity would you suggest?
Basically mirroring the `run` function? `mono_train_args`, `mono_recog_args`, ...
since `recog_args` can be reused for different steps!?